fix: update excerices and related parts (#27)
SAP-samples/cloud-sdk-team-calendar: exercises/01-getting-started.md

## Setup the destinations environment variable

Replace the URL placeholders for the `S4HANA` and `SFSF` destinations in the [`.env`](../.env) file with the URL(s) of your mock server or your SAP S/4HANA and SAP SuccessFactors systems. If you are using the mock server, the URLs for both destinations are the same, because the mock server acts as both, a mocked SAP S4/HANA system and a mocked SAP SuccessFactors system. The mock server has no authentication to access it by default. Please set the authentication type `NoAuthentication` when you use the mock server.

> ### How to find your mock server's URL
>
> Open the [SAP BTP Cockpit](https://account.hana.ondemand.com) and open the subaccount to which the mockserver was deployed. Open the space, where the mock server was deployed. You should find the URL to the mock server here.

## Start the application locally

Open the project in your IDE. For the remainder of this tutorial, we will assume you are using Visual Studio Code. Please adapt accordingly if you are using a different IDE.

> To open the project go to **File** > **Open...** and select the folder you just cloned from GitHub.

Reviewer: ```suggestion
To open the project, go to **File > Open Folder...** and select the folder you just cloned from GitHub.
```

[req] The nav text should match exactly.

> Let's check that everything works and run the application locally. First, open the command line with **Terminal** > **New Terminal**. Then, install the dependencies:

Reviewer: ```suggestion
Let's check that everything works and run the application locally. First, open the command line with **Terminal > New Terminal**. Then, install the dependencies:
```

# Getting started

> In the following, we describe how to get started with this project.

Reviewer 2: [pp] This sentence feels redundant - The header is called `getting started`

Author: I removed this sentence

You will create a new **private** repository based on this repository and run the application locally to get familiar with it.

> In the original session, the participants were handed out a participantId. If you go through the tasks on your own, make up your own id. We recommend to refrain from using personal data.

## Create a new repository based on this template

> Click the _Use this template_ button on the top of the GitHub page.

Reviewer 2: [req] Check the [style guide](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md) and ensure that the styles are corectly used. I think this is correct here, but please double check.

Author: I cahnge it along with [this style guide](https://github.com/SAP-samples/cloud-sdk-team-calendar/pull/27/files/2d490543e8066cfd789083cc6f7b09f1ca5a4b5d..61bb3a516622d6adf84b7fc2fff5a793c796cc5f)

![Use Template](images/use-template.png)

Fill in the name of your new repository (e. g. _cloud-sdk-team-calendar_) and make sure to create a **private** repository. Click on _Create repository_.
![Create New](images/create-new.png)

> ### Why private?
>
> We recommend to create your project as a private repository due to licensing constraints. In [exercise 2](02-generate-odata-clients.md) we will generate OData clients for SAP S/4HANA and SAP SuccessFactors, that is subject to SAP intellectual property and should not be made available as open source. As long as you don't publish these changes, feel free to create a public repository as well.

## Clone your repository

On your laptop open a command line and enter the following command. Don't forget to replace your username (and repository name should you have chosen a different one):

> 

Reviewer 2: [req] In the style guide there is a convention for things to be reaplced. I think `YOUR_SOME_THINGY` is the format.

Reviewer 2: [pp] Also I would remove the `Don't forget to replace your username (and repository name should you have chosen a different one)`. If someone does not understand that the username and repo need to be replaced it is hopeless anyway.

Author: I changed that along with [Placeholders style guide](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#placeholders).

```sh
git clone https://github.com/<your-github-username>/cloud-sdk-team-calendar.git
```

## Deploy the SAP S/4HANA and SAP SuccessFactors mock server (Optional)

> Note: If you have access to real SAP S/4HANA Cloud and SAP SuccessFactors (SFSF) Cloud systems, you may be able to skip this step.

Follow these [instructions](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server#how-to-run-the-server) on how to set up your SAP S/4HANA and SAP SuccessFactors mock server and deploy it on the SAP Business Technology Platform (SAP BTP), Cloud Foundry environment. 

**Please checkout the branch [`mock-server-update-data`](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server-update-data) for this tutorial instead of the branch [`mock-server`](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server) as it is described in the instructions.** The branch has the latest date data as opposed to the branch `mock-server`.

## Setup the destinations environment variable

> Replace the URL placeholders for the `S4HANA` and `SFSF` destinations in the [`.env`](../.env) file with the URL(s) of your mock server or your SAP S/4HANA and SAP SuccessFactors systems. If you are using the mock server, the URLs for both destinations are the same, because the mock server acts as both, a mocked SAP S4/HANA system and a mocked SAP SuccessFactors system. The mock server has no authentication to access it by default. Please set the authentication type `NoAuthentication` when you use the mock server.

Reviewer 2: [q] @marikaner should we add a  sentence what to do if you have a real system? Like set env for local test with basic?

> ### How to find your mock server's URL
>
> Open the [SAP BTP Cockpit](https://account.hana.ondemand.com) and open the subaccount to which the mockserver was deployed. Open the space, where the mock server was deployed. You should find the URL to the mock server here.

## Start the application locally

Open the project in your IDE. For the remainder of this tutorial, we will assume you are using Visual Studio Code. Please adapt accordingly if you are using a different IDE.

To open the project go to _File > Open..._ and select the folder you just cloned from GitHub.

Let's check that everything works and run the application locally. First, open the command line with _Terminal > New Terminal_. Then, install the dependencies:

```sh
npm install
```

Deploy a local database (locally we use SQLite):

```sh
npm run cds-deploy
```

> Start the local database:

Reviewer 2: [pp] Perhaps explain that this steps add the data from the `.csv` files to the local DB connected to the CAP service.

Author: I added additional explanations of the commands

# Getting started

You will create a new **private** repository based on this repository and run the application locally to get familiar with it.

> In the original session, the participants were handed out a participantId. If you go through the tasks on your own, make up your own id. We recommend to refrain from using personal data.

## Create a new repository based on this template

> Click `Use this template` on the top of the GitHub page.

Reviewer 3: According to [our style guide](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#use-bold-format-for-names), buttons, labels etc should not use the 'code' style (backticks) but be bold (two asterisks `**Use this template**`)

Author: apply the style guide

![Use Template](images/use-template.png)

> Fill in the name of your new repository (e. g. _cloud-sdk-team-calendar_) and make sure to create a **private** repository. Click on `Create repository`.

Reviewer 3: According to [our style guide](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#use-bold-format-for-names), buttons, labels etc should not use the 'code' style (backticks) but be bold (two asterisks `**Use this template**`)

Author: You are correct italic should not be used here but this is just an example repo name. Should a make this bold too? I think a normal text is fine.

Author: apply the style guide

![Create New](images/create-new.png)

> ### Why private?
>

> > We recommend to create your project as a private repository due to licensing constraints. In [exercise 2](02-generate-odata-clients.md) we will generate OData clients for SAP S/4HANA and SAP SuccessFactors, that is subject to SAP intellectual property and should not be made available as open source. As long as you don't publish these changes, feel free to create a public repository as well.

Reviewer 3: Q: Is this still correct? We added the pre-generated clients to the repo meanwhile (with the legal notice on copyright), so I think this is not required anymore. cc @marikaner

Author: As I discussed with @marikaner and @deekshas8, I removed the notion of the private repo.

## Clone your repository

On your laptop open a command line and enter the following command.

```sh
git clone https://github.com/YOUR_GITHUB_USERNAME/cloud-sdk-team-calendar.git
```

## Deploy the SAP S/4HANA and SAP SuccessFactors mock server (Optional)

> Note: If you have access to real SAP S/4HANA Cloud and SAP SuccessFactors Cloud systems, you may be able to skip this step.

Follow these [instructions](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server#how-to-run-the-server) on how to set up your SAP S/4HANA and SAP SuccessFactors mock server and deploy it on the SAP Business Technology Platform (SAP BTP), Cloud Foundry environment. 

> **Please checkout the branch [mock-server-update-data](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server-update-data) for this tutorial instead of the branch [mock-server](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server) as it is described in the instructions.** The branch has the latest date data as opposed to the branch mock-server.

Reviewer 3: Is there a reason why we can't update the `mock-server` branch? Seems like extra effort for the user.

Author: date data in the mock-server is 2019. If we use this, a user will have to press go back button for 3 years to show that in this tutorial hence I chose the other branch that has the latest date data.

Reviewer: There is a PR open for update the master branch https://github.com/SAP/cloud-s4-sdk-book/pull/50. If this can be merged, we can adjust the instructions. @marikaner

Author: thank you for updating the mock-server branch. I removed the instruction for using mock-server-update-data

## Setup the destinations environment variable

> Replace the URL placeholders for the `S4HANA` and `SFSF` destinations in the [`.env`](../.env) file with the URL(s) of your mock server or your SAP S/4HANA and SAP SuccessFactors systems. If you are using the mock server, the URLs for both destinations are the same, because the mock server acts as both, a mocked SAP S4/HANA system and a mocked SAP SuccessFactors system. The mock server has no authentication to access it by default. Please set the authentication type `NoAuthentication` when you use the mock server.

Reviewer 3: If possible please follow the 'one sentence per line' paradigm as suggested [here](https://github.com/SAP/cloud-sdk/pull/1063/files)

Author: I improved the sentence.

> ### How to find your mock server's URL
>
> Open the [SAP BTP Cockpit](https://account.hana.ondemand.com) and open the subaccount to which the mockserver was deployed. Open the space, where the mock server was deployed. You should find the URL to the mock server here.

## Start the application locally

Open the project in your IDE. For the remainder of this tutorial, we will assume you are using Visual Studio Code. Please adapt accordingly if you are using a different IDE.

> To open the project go to `File` > `Open...` and select the folder you just cloned from GitHub.

Reviewer 3: According to [our style guide](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#use-bold-format-for-names), buttons, labels etc should not use the 'code' style

Author: apply the style guide

> Let's check that everything works and run the application locally. First, open the command line with `Terminal` > `New Terminal`. Then, install the dependencies:

Reviewer 3: According to [our style guide](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#use-bold-format-for-names), buttons, labels etc should not use the 'code' style

Author: apply the style guide

```sh
npm install
```

Deploy a local database with these steps: 

> 1. create a local database (we use SQLite)

Reviewer 3: ```suggestion
1. create a local database with SQLite
```

[Avoid using "we"](https://github.com/SAP/cloud-sdk/blob/main/STYLEGUIDE.md#avoid-using-we)

> 2. drops existing tables and views, and re-creates them according to [your CDS model](../db/data-model.cds)

Reviewer 3: ```suggestion
2. drop existing tables and views, and re-create them according to [your CDS model](../db/data-model.cds)
```

3. deploy CSV files with initial data

> In your [`package.json`](../package.json), you have already installed `@sap/cds-dk` (SAP Cloud Application Programming Model (CAP)). SAP CAP provides a command to do the 3 steps at once. deploy a local database with the command:

Reviewer 3: Don't abbreviate as 'CAP' as described [here](https://www.sapbrandtools.com/naming-center/#/requests/7815)

Author: removed the abbreviation and follow the naming-center

```sh
npm run cds-deploy
```

> Start the local database with a [custom logic](../src/team-calendar-service.ts) that we defined for the table. SAP CAP also provides a command for this step. start the local database with the command:

Reviewer 3: Don't abbreviate as 'CAP' as described [here](https://www.sapbrandtools.com/naming-center/#/requests/7815)

Author: done

Follow these [instructions](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server#how-to-run-the-server) on how to set up your SAP S/4HANA and SAP SuccessFactors mock server and deploy it on the SAP Business Technology Platform (SAP BTP), Cloud Foundry environment.

**Checkout the branch [mock-server-update-data](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server-update-data) for this tutorial instead of the branch [mock-server](https://github.com/SAP/cloud-s4-sdk-book/tree/mock-server) as it is described in the instructions.** The branch has the latest date data as opposed to the branch mock-server.

## Setup the destinations environment variable

Replace the URL placeholders for the `S4HANA` and `SFSF` destinations in the [`.env`](../.env) file with the URL(s) of your mock server or your SAP S/4HANA and SAP SuccessFactors systems.

Since the mock server acts as both a mocked SAP S4/HANA system and SAP SuccessFactors system, the URL(s) for both destinations are the same.
Mock servers do not require authentication. If you are using the mock server, set the authentication type to `NoAuthentication`.

> ### How to find your mock server's URL

> >

Author: I converted the quote section into a subsection.

> >

Reviewer 2: please consider this also for all the other `>` areas.

> >

Author: I checked all other `>` and then removed all inappropriate quotes.
